packages/fluentui/docs/src/examples/components/Button/Visual/ButtonExampleCompose.shorthand.tsx

```diff
@@ -0,0 +1,163 @@
+import { compose, Button, ButtonProps, ButtonStylesProps, Flex, Header, Provider } from '@fluentui/react-northstar';
+import { ComponentSlotStylesInput, ComponentVariablesInput, ThemeInput } from '@fluentui/styles';
+import * as React from 'react';
+
+//
+// Components
+//
+
+// Adds a custom design term
+//
+
+type TertiaryButtonProps = {
+  tertiary?: boolean;
+};
+
+type TertiaryButtonStylesProps = TertiaryButtonProps;
+
+const ComposedButton = compose<TertiaryButtonProps, TertiaryButtonStylesProps, ButtonProps, {}>(Button, {
+  className: 'ui-tertiary-button',
+  displayName: 'TertiaryButton',
+  mapPropsToStylesProps: props => ({ tertiary: props.tertiary }),
+  handledProps: ['tertiary'],
+});
+
+// Adds overrides for a design term
+//
+
+const CompactTertiaryButton = compose(ComposedButton, {
+  displayName: 'CompactTertiaryButton',
+});
+
+// Composes custom button
+//
+
+type OverriddenButtonProps = {
+  fitted?: boolean;
+};
+
+type OverriddenButtonStylesProps = Required<OverriddenButtonProps>;
+
+const OverriddenButton = compose<OverriddenButtonProps, OverriddenButtonStylesProps, ButtonProps, {}>(Button, {
+  className: 'ui-overridden-button',
+  displayName: 'OverriddenButton',
+  mapPropsToStylesProps: props => ({
+    fitted: props.fitted,
+  }),
+  handledProps: ['fitted'],
+  overrideStyles: true,
+});
+
+//
+// Theme
+//
+
+type ComponentStylesProps = {
+  CompactTertiaryButton: ButtonStylesProps & TertiaryButtonProps;
+  TertiaryButton: ButtonStylesProps & TertiaryButtonProps;
+  OverriddenButton: OverriddenButtonProps;
+};
+
+type ComponentVariables = {
+  TertiaryButton: {
+    tertiaryBackground: string;
+    tertiaryBorderColor: string;
+    tertiaryColor: string;
+  };
+  CompactTertiaryButton: {
+    tertiaryPadding: string;
+  };
+};
+
+const componentStyles: {
+  // @ts-ignore TODO fix type there
+  [C in keyof ComponentStylesProps]: ComponentSlotStylesInput<ComponentStylesProps[C], ComponentVariables[C]>;
+} = {
+  TertiaryButton: {
+    root: ({ props: p, variables: v }) => ({
+      ...(p.tertiary && {
```

Comment:
We are basically breaking the disabled and any other state related colors that we have specified before with this. Can we update the styles to apply these colors if it is not disabled? Apart from this I don't have better idea of how to handle this :( This means that we will have always some dependency on the base styles that we are composing, but I guess it is expected. What do you think @layershifter ?

Reply:
Yep, you have to resolve styles priorities even for CSS. So any combination of design terms should be predefined

Reply:
I understand, but in CSS you can directly write the new class in the right order, here you don't have that option.. I mention this because `disabled` is a great example of a prop that you expect to be defined in the initial styles, and tertiary is not something very common.